test/helpers.mjs

```diff
@@ -29,3 +32,63 @@ export function buildUrl(path, paramObject) {
   if (params.size) result = `${result}?${params}`;
   return `https://${result}`;
 }
+
+export function logStringBytes(str1, str2) {
+  let max = Math.max(str1.length, str2.length);
+  for (let i = 0; i < max; i++) {
+    const charCode1 = str1.charCodeAt(i);
+    const charCode2 = str2.charCodeAt(i);
+    // eslint-disable-next-line no-console
+    console.log(`Byte ${i}: ${charCode1}, ${charCode2}`);
+  }
+}
+
+/**
+ * for use when non-printing characters cause test comparisons to fail. use like:
+ *
+ * logDifferentStringBytes(
+ * container.resolve("stdoutStream").getWritten(),
+ * `Type Ctrl+D or .exit to exit the shell${prompt}Database.all()\r${EOL}${stringifiedObj}${prompt}`,
+ * );
+ */
+export function logDifferentStringBytes(str1, str2) {
+  let max = Math.max(str1.length, str2.length);
+  for (let i = 0; i < max; i++) {
+    const charCode1 = str1.charCodeAt(i);
+    const charCode2 = str2.charCodeAt(i);
+    if (charCode1 !== charCode2)
+      // eslint-disable-next-line no-console
+      console.log(`Byte ${i}: ${charCode1}, ${charCode2}`);
+  }
+}
+
+/**
+ * Class representing a no-frills writable stream that, instead of writing data
+ * to a destination, holds it in memory. Can be queried for this data later;
+ * use it for testing interfaces that use streams.
+ */
+export class InMemoryWritableStream extends Writable {
```

Comment:
Test writable stream. For readable streams, I've just been using `Readable.from()` and then using the native methods on it.